src/intelhex.rs

```diff
@@ -100,9 +100,17 @@ mod test {
 
     #[test]
     fn test_hex_convert() {
-        let file = PathBuf::from("tmp/test.hex");
+        let file = ":1000000001000000000000000200000000000000ED
```

Comment:
should this be prefixed by a `b` such that it is not null terminated `b""`

Reply:
A `String` is not null terminated, so I used that one. A byte string literal can be a non-utf8 string, i.e. the conversion might fail. (I hope I got that right)